tests/pods_test.go

```diff
@@ -1,16 +1,96 @@
 package tests
 
 import (
+	"context"
+	"fmt"
+	"github.com/onsi/gomega/types"
+	"strings"
+
 	. "github.com/onsi/ginkgo"
 	. "github.com/onsi/gomega"
-	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/k8snetworkplumbingwg/kubemacpool/pkg/names"
 )
 
+const defaultNumberOfReplicas = 2
+
 var _ = Describe("Pods", func() {
-	Context("Check the client", func() {
-		It("should not fail", func() {
-			_, err := testClient.KubeClient.CoreV1().Pods("").List(v1.ListOptions{})
+	Context("Check the pod mutating webhook", func() {
+		AfterEach(func() {
+			// Clean pods from our test namespaces after every test to start clean
+			for _, namespace := range []string{TestNamespace, OtherTestNamespace} {
+				podList := &corev1.PodList{}
+				err := testClient.VirtClient.List(context.TODO(), &client.ListOptions{Namespace: namespace}, podList)
+				Expect(err).ToNot(HaveOccurred())
+
+				for _, podObject := range podList.Items {
+					err = testClient.VirtClient.Delete(context.TODO(), &podObject)
+					Expect(err).ToNot(HaveOccurred())
+				}
+
+				Eventually(func() int {
+					podList := &corev1.PodList{}
+					err := testClient.VirtClient.List(context.TODO(), &client.ListOptions{Namespace: namespace}, podList)
+					Expect(err).ToNot(HaveOccurred())
+					return len(podList.Items)
+
+				}, timeout, pollingInterval).Should(Equal(0), fmt.Sprintf("failed to remove all pod objects from namespace %s", namespace))
+
+				// This function remove all the labels from the namespace
+				err = cleanNamespaceLabels(namespace)
+			}
+
+			// Restore the default number of managers
+			err := changeManagerReplicas(defaultNumberOfReplicas)
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		testCriticalNamespace := func(namespace, label string, matcher types.GomegaMatcher) {
+			err := changeManagerReplicas(0)
+			Expect(err).ToNot(HaveOccurred())
+
+			err = addLabelsToNamespace(OtherTestNamespace, map[string]string{label: "0"})
+
+			podObject := CreatePodObject(nil)
+
+			Eventually(func() bool {
+				_, err := testClient.KubeClient.CoreV1().Pods(OtherTestNamespace).Create(podObject)
+				if err != nil && strings.Contains(err.Error(), "connection refused") {
+					return false
+				}
+
+				return true
+			}, timeout, pollingInterval).Should(matcher, "failed to apply the new pod object")
+		}
+		It("should create a pod when mac pool is running in a regular namespace", func() {
+			err := setRange(rangeStart, rangeEnd)
```

Comment:
This should be in BeforeEach, it is preparation of the context, not subject of our testing.

Reply:
even if we do it only for this `It` ?

Reply:
For me it is more readable then. But I leave it up to you. I feel it is just a matter of taste.

Reply:
we use it this way on the virtual machine tests so I will like it for this PR we can create a follow up PR to change this is necessary

Reply:
Ack, +1